### ISSUE TITLE
Error on multiplay server

### ISSUE THREAD

**Issue by KatatsumuriPan** [strubium/TheOneSmeagle#13]
The server log:
```
[HH:MM:SS] [Server thread/FATAL] [net.minecraft.server.MinecraftServer]: Error executing task
java.util.concurrent.ExecutionException: java.lang.NoSuchMethodError: mcjty.theoneprobe.items.ModItems.hasAProbeSomewhere(Lnet/minecraft/entity/player/EntityPlayer;)Z
	at java.util.concurrent.FutureTask.report(FutureTask.java:122) ~[?:1.8.0_422]
	at java.util.concurrent.FutureTask.get(FutureTask.java:192) ~[?:1.8.0_422]
	at net.minecraft.util.Util.func_181617_a(SourceFile:47) [h.class:?]
	at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:723) [MinecraftServer.class:?]
	at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:397) [nz.class:?]
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:668) [MinecraftServer.class:?]
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:526) [MinecraftServer.class:?]
	at java.lang.Thread.run(Thread.java:750) [?:1.8.0_422]
Caused by: java.lang.NoSuchMethodError: mcjty.theoneprobe.items.ModItems.hasAProbeSomewhere(Lnet/minecraft/entity/player/EntityPlayer;)Z
	at mcjty.theoneprobe.network.PacketGetInfo.getProbeInfo(PacketGetInfo.java:123) ~[PacketGetInfo.class:?]
	at mcjty.theoneprobe.network.PacketGetInfo.access$600(PacketGetInfo.java:38) ~[PacketGetInfo.class:?]
	at mcjty.theoneprobe.network.PacketGetInfo$Handler.handle(PacketGetInfo.java:113) ~[PacketGetInfo$Handler.class:?]
	at mcjty.theoneprobe.network.PacketGetInfo$Handler.lambda$onMessage$0(PacketGetInfo.java:106) ~[PacketGetInfo$Handler.class:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) ~[?:1.8.0_422]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[?:1.8.0_422]
	at net.minecraft.util.Util.func_181617_a(SourceFile:46) ~[h.class:?]
	... 5 more
```
when watching a block.

**Comment by KatatsumuriPan** [comment]
MC: 1.12.2
Mods: TheOneSmeagle only

**Comment by strubium** [comment]
I have attempted to fix this in [this](https://github.com/strubium/TheOneSmeagle/commit/c2c10759d49dbcd7a2acdb8b4e77ff43013e194b) commit. issues like this sometimes slip through because I don't have a server to test on.

**Comment by KatatsumuriPan** [comment]
I think it is good to have a local Forge server, since I will report another complex server-side issue with the mod.

**Comment by Xikaro** [comment]
When starting the server in 1.0.3, an error occurred
```
java.lang.NoSuchMethodError: net.minecraft.item.ItemArmor$ArmorMaterial.func_179242_c()Ljava/lang/String;
	at mcjty.theoneprobe.items.ModItems.getBaseTexture(ModItems.java:90)
	at mcjty.theoneprobe.items.ModItems.makeHelmet(ModItems.java:68)
	at mcjty.theoneprobe.items.ModItems.init(ModItems.java:49)
	at mcjty.theoneprobe.setup.Registration.registerItems(Registration.java:30)
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler_4_Registration_registerItems_Register.invoke(.dynamic)
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:90)
	at net.minecraftforge.fml.common.eventhandler.EventBus$1.invoke(EventBus.java:144)
	at net.minecraftforge.fml.common.eventhandler.EventBus.post(EventBus.java:182)
	at net.minecraftforge.registries.GameData.fireRegistryEvents(GameData.java:850)
	at net.minecraftforge.fml.common.Loader.preinitializeMods(Loader.java:630)
	at net.minecraftforge.fml.server.FMLServerHandler.beginServerLoading(FMLServerHandler.java:99)
	at net.minecraftforge.fml.common.FMLCommonHandler.onServerStart(FMLCommonHandler.java:333)
	at net.minecraft.server.dedicated.DedicatedServer.func_71197_b(DedicatedServer.java:125)
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:486)
	at java.lang.Thread.run(Thread.java:750)
	```

**Comment by KatatsumuriPan** [comment]
I reproduced ↑

**Comment by strubium** [comment]
Ah, annoyingly #getName() can only be called from the client

**Comment by strubium** [comment]
@Xikaro Fixed in 1.0.4 :)

**Comment by KatatsumuriPan** [comment]
![image](https://github.com/user-attachments/assets/1b83837d-bfdc-4684-88c7-7c6e3b2add67)
in 1.0.4

**Comment by KatatsumuriPan** [comment]
```
[21:33:46] [Server thread/DEBUG] [theoneprobe]: The One Probe catched error: 
java.lang.NoClassDefFoundError: net/minecraft/client/Minecraft
	at mcjty.theoneprobe.Tools.<clinit>(Tools.java:30) ~[Tools.class:?]
	at mcjty.theoneprobe.apiimpl.providers.DefaultProbeInfoProvider.showStandardBlockInfo(DefaultProbeInfoProvider.java:315) ~[DefaultProbeInfoProvider.class:?]
	at mcjty.theoneprobe.apiimpl.providers.DefaultProbeInfoProvider.addProbeInfo(DefaultProbeInfoProvider.java:57) ~[DefaultProbeInfoProvider.class:?]
	at mcjty.theoneprobe.network.PacketGetInfo.getProbeInfo(PacketGetInfo.java:148) ~[PacketGetInfo.class:?]
	at mcjty.theoneprobe.network.PacketGetInfo.access$600(PacketGetInfo.java:38) ~[PacketGetInfo.class:?]
	at mcjty.theoneprobe.network.PacketGetInfo$Handler.handle(PacketGetInfo.java:113) ~[PacketGetInfo$Handler.class:?]
	at mcjty.theoneprobe.network.PacketGetInfo$Handler.lambda$onMessage$0(PacketGetInfo.java:106) ~[PacketGetInfo$Handler.class:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_422]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_422]
	at net.minecraft.util.Util.func_181617_a(SourceFile:46) [h.class:?]
	at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:723) [MinecraftServer.class:?]
	at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:397) [nz.class:?]
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:668) [MinecraftServer.class:?]
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:526) [MinecraftServer.class:?]
	at java.lang.Thread.run(Thread.java:750) [?:1.8.0_422]
Caused by: java.lang.ClassNotFoundException: net.minecraft.client.Minecraft
	at net.minecraft.launchwrapper.LaunchClassLoader.findClass(LaunchClassLoader.java:191) ~[launchwrapper-1.12.jar:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:419) ~[?:1.8.0_422]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:352) ~[?:1.8.0_422]
	... 15 more
Caused by: net.minecraftforge.fml.common.asm.ASMTransformerWrapper$TransformerException: Exception in class transformer net.minecraftforge.fml.common.asm.transformers.SideTransformer@28c0b664 from coremod FMLCorePlugin
	at net.minecraftforge.fml.common.asm.ASMTransformerWrapper$TransformerWrapper.transform(ASMTransformerWrapper.java:260) ~[forge-1.12.2-14.23.5.2860.jar:?]
	at net.minecraft.launchwrapper.LaunchClassLoader.runTransformers(LaunchClassLoader.java:279) ~[launchwrapper-1.12.jar:?]
	at net.minecraft.launchwrapper.LaunchClassLoader.findClass(LaunchClassLoader.java:176) ~[launchwrapper-1.12.jar:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:419) ~[?:1.8.0_422]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:352) ~[?:1.8.0_422]
	... 15 more
Caused by: java.lang.RuntimeException: Attempted to load class bib for invalid side SERVER
	at net.minecraftforge.fml.common.asm.transformers.SideTransformer.transform(SideTransformer.java:62) ~[forge-1.12.2-14.23.5.2860.jar:?]
	at net.minecraftforge.fml.common.asm.ASMTransformerWrapper$TransformerWrapper.transform(ASMTransformerWrapper.java:256) ~[forge-1.12.2-14.23.5.2860.jar:?]
	at net.minecraft.launchwrapper.LaunchClassLoader.runTransformers(LaunchClassLoader.java:279) ~[launchwrapper-1.12.jar:?]
	at net.minecraft.launchwrapper.LaunchClassLoader.findClass(LaunchClassLoader.java:176) ~[launchwrapper-1.12.jar:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:419) ~[?:1.8.0_422]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:352) ~[?:1.8.0_422]
	... 15 more
```

**Comment by Krutoy242** [comment]
Hello, im developer of E2EE modpack.
Im trying to migrate from HWILA and have same error

![image](https://github.com/user-attachments/assets/571b7b86-3fcb-46b0-879c-85d3c4c90d93)
![image](https://github.com/user-attachments/assets/c9aba600-b480-430b-b6ba-5fa357648893)
![image](https://github.com/user-attachments/assets/2eb899f7-8819-44b4-844a-43ff1d481a9c)

```less
[Server thread/DEBUG] [theoneprobe]: The One Probe catched error: 
java.lang.NoClassDefFoundError: net/minecraft/client/Minecraft
	at mcjty.theoneprobe.Tools.<clinit>(Tools.java:30) ~[Tools.class:?]
	at mcjty.theoneprobe.apiimpl.providers.DefaultProbeInfoProvider.showStandardBlockInfo(DefaultProbeInfoProvider.java:315) ~[DefaultProbeInfoProvider.class:?]
	at mcjty.theoneprobe.apiimpl.providers.DefaultProbeInfoProvider.addProbeInfo(DefaultProbeInfoProvider.java:57) ~[DefaultProbeInfoProvider.class:?]
	at p455w0rd.danknull.integration.TOP$DankNullProbeInfoProvider.addProbeInfo(TOP.java:127) ~[TOP$DankNullProbeInfoProvider.class:?]
	at mcjty.theoneprobe.network.PacketGetInfo.getProbeInfo(PacketGetInfo.java:148) ~[PacketGetInfo.class:?]
```

[debug.log](https://github.com/user-attachments/files/18270265/debug.log)

**Comment by strubium** [comment]
this has been fixed in 1.0.5
